src/ProgressOnderwijsUtils/Win32/VirusScan.cs

```diff
@@ -0,0 +1,55 @@
+﻿#nullable disable annotations
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.InteropServices;
+using Microsoft.Windows.Sdk;
+using ProgressOnderwijsUtils;
+
+namespace ProgressOnderwijsUtils.Win32
+{
+    public static class VirusScan
+    {
+        public static bool IsMalware(string content, string contentName)
+        {
+            using var scanSession = new VirusScanSessie("Progress.Net");
+            PInvoke.AmsiScanString(scanSession.Context, content, contentName, IntPtr.Zero, out var result);
+            return ResultIsMalware(result);
+        }
+
+        public static bool IsMalware(byte[] buffer, string contentName) => IsMalwareUnsafe(buffer, contentName);
+
+        static unsafe bool IsMalwareUnsafe(byte[] buffer, string contentName)
+        {
+            var sizet = Marshal.SizeOf(typeof(byte)) * buffer.Length;
+            var bufferPtr = Marshal.AllocHGlobal(sizet).ToPointer();
+
+            using var scanSession = new VirusScanSessie("Progress.Net");
+            PInvoke.AmsiScanBuffer(scanSession.Context, bufferPtr, (uint)buffer.LongLength, contentName, IntPtr.Zero, out var result);

```

Comment:
Je geeft je nieuwe buffer door, maar met de lengte van de oude buffer zonder sizeof scaling (gaat hier goed, want byte, maar tricky)

...maar je nieuwe buffer is nooit gevuld, dus dit zal *N* zero-bytes scannen.